id_test.go

```diff
@@ -0,0 +1,71 @@
+package mongo_test
+
+import (
+	"regexp"
+	"testing"
+
+	"gopkg.in/mgo.v2/bson"
+
+	mongo "github.com/rs/rest-layer-mongo"
+)
+
+const (
+	validObjectID   = "59a40602952dbd0001c3ffc9"
+	invalidObjectID = "59a40602952dbd0001c3ffc9f"
+)
+
+func TestObjectIDValidate(t *testing.T) {
+	v := &mongo.ObjectID{}
+
+	t.Run("validObjectID", func(t *testing.T) {
+		expect := bson.ObjectIdHex(validObjectID)
+		id, err := v.Validate(validObjectID)
+		if expect != id {
+			t.Errorf("v.Validate(validObjectID):\n %v (expect) != %v (actual)", expect, id)
+		}
+		if err != nil {
+			t.Error("v.Validate(validObjectID):\n unexpected error:", err)
+		}
+	})
+
+	t.Run("invalidObjectID", func(t *testing.T) {
+		id, err := v.Validate(invalidObjectID)
+		if nil != id {
+			t.Errorf("v.Validate(invalidObjectID):\n %v (expect) != %v (actual)", nil, id)
+		}
+		if err == nil {
+			t.Error("v.Validate(invalidObjectID):\n expected error, got nil")
+		}
+	})
+}
+
+func TestObjectIDJSONSchmea(t *testing.T) {
```

Comment:
You can compare this one to the `testify/assert` version and see which one you like best.